### PR TITLE
02) User Permissions

### PR DESCRIPTION
**Pull request by crudobaker**

### CODE REVIEW

File: bot/src/telegram/actions/assign-guard.js [hunk 1]
```diff
@@ -33,4 +34,4 @@ const configure = (bot) => {
   bot.addAction(name, assignGuard)
 }
 
-export const ASSIGN_GUARD = { name, configure }
+export const ASSIGN_GUARD = { name, configure, permissions: [COORDINATOR_ROLE] }
```
Comment (pablotocalini) [inline bot/src/telegram/actions/assign-guard.js:37]: where did "permissions" appear from?

File: bot/src/telegram/actions/create-guard.js [hunk 2]
```diff
@@ -2,9 +2,10 @@ import agenda from 'bot/src/init.js'
 import { formatDate } from 'core/src/date-utils.js'
 import { GUARD_ALREADY_EXISTS, GUARD_DATE_IS_PAST } from 'core/src/model/agenda.js'
 import { confirmActionKeyboard, readCallbackQueryParams } from 'bot/src/telegram/keyboard-utils.js'
+import { COORDINATOR_ROLE } from 'core/src/model/user-role.js'
 
 const askForConfirmation = (ctx, date) => {
-  const options = confirmActionKeyboard(createGuardAction, date.getTime())
+  const options = confirmActionKeyboard(ctx, createGuardAction, date.getTime())
```
Comment (pablotocalini) [inline bot/src/telegram/actions/create-guard.js:8]: why do you need the "ctx" here?

File: bot/src/telegram/commands/index.js [hunk 3]
```diff
@@ -3,12 +3,20 @@ import { MY_NEXT_GUARD } from 'bot/src/telegram/commands/my-next-guard-cmd.js'
...
+  COMMANDS.forEach((command) => bot.command(command.name, command.apply))
+}
+
+export function getCommandsForUser(user) {
+  //a command is for a user if the command doesn't require permissions or the
+  //permission list includes the user role.
+  return COMMANDS.filter((command) => !command.permissions.length || command.permissions.includes(user.role))
+}
+
+export function canExecuteCommand(user, ctx) {
```
Comment (pablotocalini) [inline bot/src/telegram/commands/index.js:18]: I need three weeks to understand this lines

**Review by pablotocalini — APPROVED** [review]

File: bot/src/telegram/actions/create-guard.js (hunk 2 above)
Comment (crudobaker) [reply bot/src/telegram/actions/create-guard.js:8]: This a a new param we need to pass to the keyboard functions to create or not the buttons, depending on the user/action permissions.

**Pull request merged**